backend/src/channels/channels.service.ts

```diff
@@ -7,22 +7,58 @@ import { CreateChannelDto } from './dto/create-channel.dto';
 import { QueryChannelDto } from './dto/query-channel.dto';
 import { UpdateChannelDto } from './dto/update-channel.dto';
 import { ResponseChannelDto } from './dto/response-channel.dto';
-import { Channel } from './entities/channel.entity';
+import { Channel, ChannelType } from './entities/channel.entity';
 import ChannelRepository from './repository/channel.repository';
+import UserRepository from 'src/users/repository/user.repository';
 
 @Injectable()
 export class ChannelsService {
   constructor(
     @InjectRepository(ChannelRepository)
     private channelsRepository: ChannelRepository,
+    @InjectRepository(UserRepository)
+    private usersRepository: UserRepository,
   ) {}
 
-  create(createChannelDto: CreateChannelDto) {
+  async create(createChannelDto: CreateChannelDto) {
     const channel: Channel = new Channel();
-    channel.name = createChannelDto.name;
     channel.type = createChannelDto.type;
-    channel.password = createChannelDto.password;
-    return this.channelsRepository.save(channel);
+    if (channel.type == ChannelType.DIRECT)
+      await this.checkDirectUsers(createChannelDto, channel);
+    else {
+      if (createChannelDto.name[0] == '-')
+        throw "Channel name cannot start with a '-'.";
+      channel.name = createChannelDto.name;
+      channel.password = createChannelDto.password;
+    }
+    return await this.channelsRepository.save(channel);
+  }
+
+  async checkDirectUsers(createChannelDto: CreateChannelDto, channel: Channel) {
+    if (createChannelDto.userOneId > createChannelDto.userTwoId)
+      [createChannelDto.userOneId, createChannelDto.userTwoId] = [
+        createChannelDto.userTwoId,
+        createChannelDto.userOneId,
+      ];
+
+    channel.name =
+      '-' + createChannelDto.userOneId + '-' + createChannelDto.userTwoId;
+    if (
+      await this.channelsRepository.findOne({
+        where: {
+          type: 'direct',
```

Comment:
```suggestion
          type: ChannelType.DIRECT,
```
I would recommend using the enum so that we only have to change the value in one place. Unless this breaks the `where` query